src/Symfony/Bridge/Doctrine/Tests/Decorator/DoctrineTransactionDecoratorTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Tests\Decorator;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+use PHPUnit\Framework\TestCase;
+use Symfony\Bridge\Doctrine\Attribute\Transactional;
+use Symfony\Bridge\Doctrine\Decorator\DoctrineTransactionDecorator;
+use Symfony\Component\Decorator\DecoratorChain;
+
+class DoctrineTransactionDecoratorTest extends TestCase
+{
+    private ManagerRegistry $managerRegistry;
+    private Connection $connection;
+    private EntityManagerInterface $entityManager;
+    private DecoratorChain $decorator;
+
+    protected function setUp(): void
+    {
+        $this->connection = $this->createMock(Connection::class);
+
+        $this->entityManager = $this->createMock(EntityManagerInterface::class);
+        $this->entityManager->method('getConnection')->willReturn($this->connection);
+
+        $this->managerRegistry = $this->createMock(ManagerRegistry::class);
+        $this->managerRegistry->method('getManager')->willReturn($this->entityManager);
+
+        $this->decorator = DecoratorChain::from([
+            DoctrineTransactionDecorator::class => fn () => new DoctrineTransactionDecorator($this->managerRegistry),
+        ]);
+    }
+
+    public function testDecoratorWrapsInTransactionAndFlushes()
+    {
+        $this->connection->expects($this->once())
+            ->method('beginTransaction')
+        ;
+        $this->connection->expects($this->once())
+            ->method('commit')
+        ;
+        $this->entityManager->expects($this->once())
+            ->method('flush')
+        ;
+
+        $handler = new TestHandler();
+        $result = $this->decorator->call($handler->handle(...));
+        $this->assertSame('success', $result);
+    }
+
+    public function testTransactionIsRolledBackOnException()
+    {
+        $this->connection->expects($this->once())
+            ->method('beginTransaction')
+        ;
+        $this->connection->expects($this->once())
+            ->method('rollBack')
+        ;
+
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage('A runtime error.');
+
+        $handler = new TestHandler();
+        $this->decorator->call($handler->handleWithError(...));
+    }
+
+    public function testInvalidEntityManagerThrowsException()
+    {
+        $this->managerRegistry
+            ->method('getManager')
+            ->with('unknown_manager')
+            ->will($this->throwException(new \InvalidArgumentException()));
```

Comment:
I think this notation is deprecated, you should use `willThrowException()` instead